File: backend/src/main/java/com/voltpay/backend/repository/PaymentRequestRepository.java
package com.voltpay.backend.repository;

import com.voltpay.backend.model.PaymentRequest;
import org.springframework.data.mongodb.repository.MongoRepository;
import java.util.List;

public interface PaymentRequestRepository extends MongoRepository<PaymentRequest, String> {
    List<PaymentRequest> findByPayerPhoneAndStatus(String payerPhone, String status);
    List<PaymentRequest> findByRequesterPhone(String requesterPhone);
}

File: backend/src/main/kotlin/com/voltpay/backend/repository/PaymentRequestRepository.kt
package com.voltpay.backend.repository

import com.voltpay.backend.model.PaymentRequest
import org.springframework.data.mongodb.repository.MongoRepository

interface PaymentRequestRepository : MongoRepository<PaymentRequest, String> {
    fun findByPayerPhoneAndStatus(payerPhone: String, status: String): List<PaymentRequest>
    fun findByRequesterPhone(requesterPhone: String): List<PaymentRequest>
}
